code/menuui/techmenu.cpp

```diff
@@ -235,14 +236,10 @@ typedef struct {
 	int textures_loaded;	// if the model has textures loaded for it or not (hacky mem management)
 } tech_list_entry;
 
-static tech_list_entry *Ship_list = nullptr;
-static int Ship_list_size = 0;
-static tech_list_entry *Weapon_list = nullptr;
-static int Weapon_list_size = 0;
-static tech_list_entry *Intel_list = nullptr;
-static int Intel_list_size = 0;
-static tech_list_entry *Current_list;								// points to currently valid display list
-static int Current_list_size = 0;
+SCP_vector<tech_list_entry> Ship_list;
+SCP_vector<tech_list_entry> Weapon_list;
+SCP_vector<tech_list_entry> Intel_list;
+SCP_vector<tech_list_entry> Current_list;								// A copy of the currently valid display list
```

Comment:
`Current_list` should just be a ptr to the active list. Unless I'm missing something, there really isn't a reason to duplicate that info.

Additionally, bitmaps and animations are loaded into `Current_list` instead of the originals, and never released. Another reason it should just point to the appropriate list.

Reply:
Agreed, `Current_list` should be a pointer.

Reply:
Alright, made the change, built it and it runs.